pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -17,20 +17,168 @@
+type StoragePoolStatus struct {
+	// ReferenceResource points to the pre-existing pool resource or the created pool resource after creation of pool.
+	// +optional
+	ReferenceResource corev1.ObjectReference `json:"referenceResource,omitempty"`
+
+	// Capacity of the pool,viz total, used and available capacity
+	// +optional
+	Capacity StorageCapacity `json:"capacity,omitempty"`
+
+	// IOPs of the pool,viz total, provisioned, used and available IOPs
+	// +optional
+	IOPs StorageIOPs `json:"IOPs,omitempty"`
+
+	// VolumeSizeMaxLimit for maximum volume size allowed
+	// +optional
+	VolumeSizeMaxLimit resource.Quantity `json:"volumeSizeMaxLimit,omitempty"`
+
+	// Conditions enlist the conditions of the pool based on various types.
+	// +optional
+	Conditions []StoragePoolCondition `json:"state,omitempty"`
+}
+
+// StorageCapacity for the pool,viz total, used and available capacity
+type StorageCapacity struct {
+	// Total Capacity of the pool
+	// +optional
+	Total resource.Quantity `json:"total,omitempty"`
+
+	// Used Capacity of the pool
+	// +optional
+	Used resource.Quantity `json:"used,omitempty"`
+
+	// Available Capacity of the pool
+	// +optional
+	Available resource.Quantity `json:"available,omitempty"`
+
+	// Provisioned Capacity of the pool
+	// +optional
+	Provisioned resource.Quantity `json:"provisioned,omitempty"`
+}
+
+// StorageIOPs for the pool,viz total, provisioned, used and available capacity
+type StorageIOPs struct {
+	// Total IOPs of the pool
+	// +optional
+	Total uint64 `json:"total,omitempty"`
+
+	// Available IOPs of the pool
+	// +optional
+	Available uint64 `json:"available,omitempty"`
+
+	// Provisioned IOPs of the pool
+	// +optional
+	Provisioned uint64 `json:"provisioned,omitempty"`
+
+	// Used IOPs of the pool
+	// +optional
+	Used uint64 `json:"used,omitempty"`
+}
+
+// StoragePoolCondition describes the condition of a StoragePool at a certain point based on a specific type
+type StoragePoolCondition struct {
+	// Type of StoragePool condition.
+	// +optional
+	Type StoragePoolConditionType `json:"type"`
+
+	// Status of the condition, one of True, False, Unknown.
+	// +optional
+	Status corev1.ConditionStatus `json:"status"`
+
+	// The last time this condition was updated.
+	// +optional
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty"`
+
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// The reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// A human readable message indicating details about the transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// StoragePoolConditionType specifies the particular state that the conditions are based on.
+type StoragePoolConditionType string
+
+const (
+	// StoragePoolConditionTypePoolExpansion condition will be available when user triggers
```

Comment:
Will there be multiple conditions assigned for a pool at given type (as i see we are specifying list in spec)? I think all the following condition types are mutually exclusive. Do we have state transition diagram defined for below states?

Reply:
Valid point. Maybe we should only have bare minimal states here (as the scheduler is going to use the information).
PoolOnline, PoolOffline, PoolDegraded.
@shovanmaity your thoughts?